runtime/src/accounts.rs

```diff
@@ -96,20 +93,29 @@ impl From<usize> for AccountStorageStatus {
     }
 }
 
-struct AccountIndexInfo {
-    /// For each Pubkey, the account for a specific fork is in a specific
-    /// AppendVec at a specific index
-    index: RwLock<HashMap<Pubkey, AccountMap>>,
+// in a given a Fork, which AppendVecId and offset
+type AccountMap = RwLock<HashMap<Fork, (AppendVecId, u64)>>;
+
+/// information about where Accounts are stored and which vote accounts are present
+/// keying hierarchy is:
+///
+///    pubkey->fork->append_vec->offset
+///
+struct AccountIndex {
+    /// For each Pubkey, the Account for a specific Fork is in a specific
+    ///  AppendVec at a specific index.  There may be an Account for Pubkey
+    ///  in any number of Forks.
+    account_maps: RwLock<HashMap<Pubkey, AccountMap>>,
 
     /// Cached index to vote accounts for performance reasons to avoid having
-    /// to iterate through the entire accounts each time
-    vote_index: RwLock<HashSet<Pubkey>>,
+    ///  to iterate through the entire accounts each time
+    vote_accounts: RwLock<HashSet<Pubkey>>,
 }
 
 /// Persistent storage structure holding the accounts
 struct AccountStorage {
     /// storage holding the accounts
-    appendvec: Arc<RwLock<AppendVec<Account>>>,
+    accounts: Arc<RwLock<AppendVec<Account>>>,
```

Comment:
The accounts name is a bit overused, but agree that appendvec isn't great either.

Reply:
I have some plans for "accounts" ;)